src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -645,7 +654,10 @@ Pipe StorageRabbitMQ::read(
     }
 
     if (!connection->getHandler().loopRunning() && connection->isConnected())
+    {
+        connection->getHandler().updateLoopState(Loop::RUN);
```

Comment:
Could you please add a comment about the race condition with the explanation why we don't care?